File: nearbyfundi_backend/app/Models/Service.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsToMany;

class Service extends Model
{
    protected $fillable = ['name', 'swahili_name'];

    protected $casts = [
        'created_at' => 'datetime',
        'updated_at' => 'datetime',
    ];

    /**
     * Get the technicians for this service.
     */
    public function technicians(): BelongsToMany
    {
        return $this->belongsToMany(Technician::class, 'technician_services');
    }

    /**
     * Get the requests for this service.
     */
    public function requests()
    {
        return $this->hasMany(ServiceRequest::class);
    }

    /**
     * Get the categories for this service.
     */
    public function categories(): BelongsToMany
    {
        return $this->belongsToMany(
            ServiceCategory::class,
            'service_category_assignments',
            'service_id',
            'service_categoryID'
        )->withTimestamps();
    }

    /**
     * Get service name in the requested language.
     */
    public function getNameForLocale(string $locale = 'en'): string
    {
        if ($locale === 'sw' && $this->swahili_name) {
            return $this->swahili_name;
        }
        return $this->name;
    }

    /**
     * Check if service has a specific category.
     * FIXED: Explicitly reference pivot table column to avoid ambiguity.
     */
    public function hasCategory($categoryId): bool
    {
        return $this->categories()
            ->where('service_category_assignments.service_categoryID', $categoryId)
            ->exists();
    }

    /**
     * Get category IDs as array.
     * FIXED: Explicitly reference pivot table column to avoid ambiguity.
     */
    public function getCategoryIdsAttribute(): array
    {
        return $this->categories()
            ->pluck('service_category_assignments.service_categoryID')
            ->toArray();
    }

    /**
     * Get category names as array.
     */
    public function getCategoryNamesAttribute(): array
    {
        return $this->categories()->pluck('category_name')->toArray();
    }

    /**
     * Sync categories for this service.
     */
    public function syncCategories(array $categoryIds): void
    {
        $this->categories()->sync($categoryIds);
    }

    /**
     * Attach a category to this service.
     */
    public function attachCategory($categoryId): void
    {
        if (!$this->hasCategory($categoryId)) {
            $this->categories()->attach($categoryId);
        }
    }

    /**
     * Detach a category from this service.
     */
    public function detachCategory($categoryId): void
    {
        $this->categories()->detach($categoryId);
    }
}

File: fundiapp/lib/models/service.dart
// lib/models/service.dart

class ServiceCategory {
  final int id;
  final String name;
  final String swahiliName;
  final String slug;

  ServiceCategory({
    required this.id,
    required this.name,
    this.swahiliName = '',
    this.slug = '',
  });

  factory ServiceCategory.fromJson(Map<String, dynamic> json) {
    return ServiceCategory(
      id: json['id'] ?? json['service_categoryID'] ?? 0,
      name: json['name'] ?? json['category_name'] ?? '',
      swahiliName: json['swahili_name'] ?? '',
      slug: json['slug'] ?? '',
    );
  }
}

class Service {
  final int id;
  final String name;
  final String swahiliName;
  final List<ServiceCategory> categories;

  Service({
    required this.id,
    required this.name,
    this.swahiliName = '',
    this.categories = const [],
  });

  factory Service.fromJson(Map<String, dynamic> json) {
    List<ServiceCategory> categories = [];
    if (json['categories'] != null && json['categories'] is List) {
      categories = (json['categories'] as List)
          .map((e) => ServiceCategory.fromJson(e as Map<String, dynamic>))
          .toList();
    }

    return Service(
      id: json['id'] ?? 0,
      name: json['name'] ?? '',
      swahiliName: json['swahili_name'] ?? '',
      categories: categories,
    );
  }
}